### PR TITLE
Build with cpp_implementation

### PR DESCRIPTION
**Pull request by hajapy**
Addresses https://github.com/conda-forge/protobuf-feedstock/issues/28

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -16,17 +16,7 @@ then
...
-# Doesn't include gmock or gtest. So, need to get these ourselves for `make check`.
-git clone -b release-1.7.0 git://github.com/google/googlemock.git gmock
-git clone -b release-1.7.0 git://github.com/google/googletest.git gmock/gtest
-
-# Build configure/Makefile as they are not present.
-aclocal
-libtoolize
-autoconf
-autoreconf -i
-automake --add-missing
```
Comment (jakirkham) [inline recipe/build.sh]: Why are `gmock` and `gtest` being dropped?

Comment (hajapy) [inline recipe/build.sh]: The autogen.sh script handle bringing them in: https://github.com/google/protobuf/blob/v3.3.0/autogen.sh

Comment (hajapy) [inline recipe/build.sh]: I noticed this way of doing things by looking at https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/ci_build/protobuf/protobuf_optimized_pip.sh

Comment (hajapy) [inline recipe/build.sh]: It's also in the Unix c++ install guide: https://github.com/google/protobuf/blob/master/src/README.md#c-installation---unix

**Comment by hajapy** [comment]
It seems `--cpp_implementation` is not supported for Windows:

Some quotes from here

https://groups.google.com/forum/#!searchin/protobuf/windows%7Csort:relevance/protobuf/bNM1lGFzabE/_NUiDqQhAgAJ
> Unfortunately, python c++ implementation is not supported on windows yet.

https://groups.google.com/forum/#!searchin/protobuf/cpp_implementation%7Csort:relevance/protobuf/EA9R3mWwJw0/Wpi1RXQ0DwAJ
> That's not supported. We only test python C++ implementation for linux and its windows version is known to be broken for quite a while. You are welcome to send us pull requests to fix the windows build issues though.

And open issues on github:
https://github.com/google/protobuf/issues/1998
https://github.com/google/protobuf/issues/975




I don't think I should go down the rabbit hole of patching setup.py here to try to get it to build.

**Comment by jakirkham** [comment]
This might have been why I punted on it last time.

So I guess the question is, do we want to keep Windows as pure Python and use C++ for Mac and Linux platforms. Thoughts from others?

**Comment by hajapy** [comment]
Well I may be biased, but I think it's fine to have the Mac/Linux platforms use cpp while Windows uses the python implementation, given that this represents the best available support provided by the protobuf project. If they ever fix the Windows build upstream, we can have cpp builds for all platforms. 

Worth noting that on pypi, they have cpp wheels for linux, but not Mac (and obviously not Windows either).

**Comment by hajapy** [comment]
@dopplershift @ocefpaf @jjhelmus ping! feedback requested on whether to go forward with cpp for linux/mac and python on win or python for all platforms (see above discussions)

**Comment by ocefpaf** [comment]
@hajapy I don't see any problem with that in principle, but I'll wait for @dopplershift and @jjhelmus to give their opinion before we merge it.

**Comment by dopplershift** [comment]
I don't have any problems with that easier--if it's good enough for the people who actually make the library, it's good enough for me here.

**Comment by hajapy** [comment]
@jakirkham can we make the call on this or are we still waiting for @jjhelmus?

**Comment by jjhelmus** [comment]
LGTM

**Pull request merged**